ffi/diplomat/src/pluralrules.rs

```diff
@@ -144,31 +123,19 @@ pub mod ffi {
     impl ICU4XPluralOperands {
         /// FFI version of `PluralOperands::from_str()`.
         #[diplomat::rust_link(icu_plurals::PluralOperands::from_str, FnInStruct)]
-        pub fn create(s: &str) -> ICU4XCreatePluralOperandsResult {
+        pub fn create(s: &str) -> DiplomatResult<ICU4XPluralOperands, ICU4XError> {
             PluralOperands::from_str(s)
-                .ok()
-                .map(|ops| ICU4XCreatePluralOperandsResult {
-                    operands: ICU4XPluralOperands {
-                        i: ops.i,
-                        v: ops.v,
-                        w: ops.w,
-                        f: ops.f,
-                        t: ops.t,
-                        c: ops.c,
-                    },
-                    success: true,
-                })
-                .unwrap_or(ICU4XCreatePluralOperandsResult {
-                    operands: ICU4XPluralOperands {
-                        i: 0,
-                        v: 0,
-                        w: 0,
-                        f: 0,
-                        t: 0,
-                        c: 0,
-                    },
-                    success: false,
+                .map(|ops| ICU4XPluralOperands {
+                    i: ops.i,
+                    v: ops.v,
+                    w: ops.w,
+                    f: ops.f,
+                    t: ops.t,
+                    c: ops.c,
                 })
+                // XXX should this have its own errors?
```

Comment:
Question for @zbraniecki, but I'd say probably not. I *think* I would rather err on the side of too few errors than too many, because we can always add more, but it's much harder to shrink the set. Might be a good question to ask the group.